### ISSUE TITLE
Generate code coverage report

### ISSUE THREAD

**Issue by stephenmichaelf** [mediumTaj/dot-net-sdk#9]
Find a free service to generate code coverage.

Potentially:

https://github.com/OpenCover/opencover

**Comment by mediumTaj** [comment]
Awesome... do you know if this is something we can run in travis?

**Comment by stephenmichaelf** [comment]
That's a good question! I can't tell. I also saw this which is widely used and free for open source, it also lets us integrate with GitHub: https://coveralls.io/features

**Comment by atilatosta** [comment]
@mediumTaj , yes you can use it with Travis. There are two ways to do it: using coveralls.io to generate the badge or using Opencover with the ReportGenerator. The latter will need to create a script that runs on Travis, to push the file in Git. I'm using OpenCover with Coveralls.io, in dotnet-standard-sdk.

**Comment by mediumTaj** [comment]
@atilatosta Awesome thanks Atila... I'll look into it. Also above is Steve (@stephenmichaelf). He's a .NET developer that will be contributing to our .NET SDKs.

**Comment by mediumTaj** [comment]
@atilatosta You are using OpenCover in Travis? Is there a version we can use in mono? I have coverage generating locally through OpenCover and a script set up to generate the summary.

**Comment by mediumTaj** [comment]
@atilatosta Do you have an example of getting OpenCover to work in Travis? I thought it was Windows only. That would be awesome or I would have to use AppVeyor as well just to generate and send code coverage!

**Comment by mediumTaj** [comment]
I am generating the report locally now. I don't think OpenCover or ReportGenerator will work in Travis. Also sending to Coveralls fails all giving me an error 

```
422 error: Couldn't find a repository matching this job
```

I have a badge on there from shields.io with my code coverage but we need to figure out a way to get OpenCover, ReportGenerator and Coveralls to work together either by switching to AppVeyor (and get documentation generation as well) or switching form OpenCover to monocov which looks like it might work in Travis.

https://github.com/csmacnz/Coveralls.net-Samples

I'm not sure why I'm getting the error on Coveralls - I've removed the repo from Coveralls, reset the token, and reconnected Coveralls to Github.
